src/Models/Model.php

```diff
@@ -149,6 +149,7 @@ protected function getPropertyDefault( string $key ) {
 	protected function getPropertyDefaults() : array {
 		$defaults = [];
 		foreach ( array_keys( $this->properties ) as $property ) {
+			// @todo This could be an edge case bug, we shouldn't assume `null` is always a valid default value.
 			$defaults[ $property ] = $this->getPropertyDefault( $property );
```

Comment:
An issue I can see happening here is a conflict between things that are explicitly being set to `NULL` versus ones that have not been set at all. This muddies that water, and makes it hard to know which situation is happening.

We ought to embrace MySQL and their `DEFAULT` and `NULL` features, and a PHP `null` would translate to a MySQL `NULL`. And a non-defined value (e.g. not set to `NULL` or any other value) should allow MySQL to leverage it's `DEFAULT` definitions.

https://dev.mysql.com/doc/refman/8.0/en/data-type-defaults.html